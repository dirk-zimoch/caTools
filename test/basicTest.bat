echo
echo "-----------------------------------------"
echo "Various output formatting"
echo "-----------------------------------------"
echo "Using: -plain"
call get.bat -plain
echo
echo "Using: -noname -nostat -nounit"
call get.bat -noname -nostat -nounit
echo
echo "Using: -stat -plain"
call get.bat -stat -plain
echo
echo "Using: -stat -nostat"
call get.bat -stat -nostat
echo
echo "Using: -d -t -localdate -localtime"
call get.bat -d -t -localdate -localtime
echo

echo
echo "-----------------------------------------"
echo "caPut on waveforms"
echo "-----------------------------------------"
echo "with -bin"
call putWf.bat -bin
echo
echo "with -oct"
call putWf.bat -oct
echo
echo "with -hex"
call putWf.bat -hex

echo
echo "-----------------------------------------"
echo "Test setting and reading enum values with number or string"
echo "-----------------------------------------"
..\caput CTT:TEST_MBBO 1
..\caput CTT:TEST_MBBO "25.00 Hz"
..\caget -num CTT:TEST_MBBO
..\caget  CTT:TEST_MBBO

..\caput CTT:BO 0
..\caput CTT:BO 1
..\caput CTT:BO 0
..\caput -s CTT:BO 0
..\caput CTT:BO 0

echo
echo "-----------------------------------------"
echo "Test setting and reading chars as numbers or strings"
echo "-----------------------------------------"
..\caput -s CTT:TEST_WVF-CHAR "1 2"
..\caget -num CTT:TEST_WVF-CHAR
..\caget CTT:TEST_WVF-CHAR
..\caput CTT:TEST_WVF-CHAR "1 2"
..\caget -num CTT:TEST_WVF-CHAR
..\caget CTT:TEST_WVF-CHAR
..\caput -num CTT:TEST_WVF-CHAR "50 32 32 32 49"
..\caget -num CTT:TEST_WVF-CHAR
..\caget CTT:TEST_WVF-CHAR

echo
echo "-----------------------------------------"
echo "Double conversion"
echo "-----------------------------------------"
..\caput CTT:CALC 10.23
..\caput CTT:CALC.PREC 4
..\caget -hex CTT:CALC
..\caget -bin CTT:CALC
..\caget -oct CTT:CALC
..\caget -s CTT:CALC
..\caget -num CTT:CALC

echo
echo "-----------------------------------------"
echo "Number of elements"
echo "-----------------------------------------"
..\caget -nord CTT:TEST_WVF-CHAR CTT:TEST_WVF-DOUBLE CTT:TEST_WVF-FLOAT CTT:TEST_WVF-LONG CTT:TEST_WVF-SHORT CTT:TEST_WVF-STRING CTT:PROC-CALC CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest.NAME
