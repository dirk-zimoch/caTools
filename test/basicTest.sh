./procCalc.sh
./getGetsDo.sh
./put.sh
./getGetsDo.sh

echo
echo "-----------------------------------------"
echo "Various output formatting"
echo "-----------------------------------------"
echo "Using: -plain"
./get.sh -plain
echo
echo "Using: -noname -nostat -nounit"
./get.sh "-noname -nostat -nounit"
echo
echo "Using: -stat -plain"
./get.sh "-stat -plain"
echo
echo "Using: -stat -nostat"
./get.sh "-stat -nostat"
echo
echo "Using: -d -t -localdate -localtime"
./get.sh "-d -t -localdate -localtime"
echo
echo "Using: -mukmuk"
./get.sh -mukmuk

echo
echo "-----------------------------------------"
echo "caPut on waveforms"
echo "-----------------------------------------"
echo "with -bin"
./putWf.sh -bin
echo
echo "with -oct"
./putWf.sh -oct
echo
echo "with -hex"
./putWf.sh -hex

echo
echo "-----------------------------------------"
echo "Test setting and reading enum values with number or string"
echo "-----------------------------------------"
../caput CTT:TEST_MBBO 1
../caput CTT:TEST_MBBO "25.00 Hz"
../caget -num CTT:TEST_MBBO
../caget  CTT:TEST_MBBO

../caput CTT:BO 0
../caput CTT:BO 1
../caput CTT:BO 0
../caput -s CTT:BO 0
../caput CTT:BO 0

echo
echo "-----------------------------------------"
echo "Test setting and reading chars as numbers or strings"
echo "-----------------------------------------"
../caput -s CTT:TEST_WVF-CHAR "1 2"
../caget -num CTT:TEST_WVF-CHAR
../caget CTT:TEST_WVF-CHAR
../caput CTT:TEST_WVF-CHAR "1 2"
../caget -num CTT:TEST_WVF-CHAR
../caget CTT:TEST_WVF-CHAR
../caput -num CTT:TEST_WVF-CHAR "50 32 32 32 49"
../caget -num CTT:TEST_WVF-CHAR
../caget CTT:TEST_WVF-CHAR

echo
echo "-----------------------------------------"
echo "Double conversion"
echo "-----------------------------------------"
../caput CTT:CALC 10.23
../caput CTT:CALC.PREC 4
../caget -hex CTT:CALC
../caget -bin CTT:CALC
../caget -oct CTT:CALC
../caget -s CTT:CALC
../caget -num CTT:CALC

echo
echo "-----------------------------------------"
echo "Number of elements"
echo "-----------------------------------------"
../caget -nord CTT:TEST_WVF-CHAR CTT:TEST_WVF-DOUBLE CTT:TEST_WVF-FLOAT CTT:TEST_WVF-LONG CTT:TEST_WVF-SHORT CTT:TEST_WVF-STRING CTT:PROC-CALC CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest.NAME
