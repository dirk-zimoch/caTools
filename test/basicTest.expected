Start this test just after first starting the test IOC
caGet output
CTT:PROC-CALC 0 (SEVR:INVALID STAT:UDF)

caGets output
CTT:PROC-CALC 1 

making caDo request
caGet output
CTT:PROC-CALC 2 

done

caGet
CTT:TEST_1-AI.SCAN ".1 second" 
CTT:TEST_1-AI.INP "CTT:TEST_1 NPP NMS" 
CTT:TEST_1-AI.EGU "ds" 
CTT:CALC_BI.INAV "Local PV" 
CTT:TEST_MBBO "50.00 Hz" (SEVR:MAJOR STAT:STATE)
CTT:TEST_MBBO.ZRVL 1 (SEVR:MAJOR STAT:STATE)
CTT:TEST_MBBO.DESC "Select RF 1 event freq" (SEVR:MAJOR STAT:STATE)
CTT:TEST_STRING "Hello world" (SEVR:INVALID STAT:UDF)
CTT:STRING "Value field can have only fourty chars." (SEVR:INVALID STAT:UDF)
CTT:STRING.OUT "ten1234567twenty1234thirty1234forty12345fifty12345sixty12345 NPP NMS" (SEVR:INVALID STAT:UDF)
CTT:STRING.DESC "Description field can have only fourty ." (SEVR:INVALID STAT:UDF)
CTT:TEST_WVF-CHAR  
CTT:TEST_WVF-DOUBLE  (SEVR:INVALID STAT:UDF)
CTT:TEST_WVF-FLOAT  (SEVR:INVALID STAT:UDF)
CTT:TEST_WVF-FLOAT.NELM 10 (SEVR:INVALID STAT:UDF)
CTT:TEST_WVF-LONG  (SEVR:INVALID STAT:UDF)
CTT:TEST_WVF-SHORT  (SEVR:INVALID STAT:UDF)
CTT:TEST_WVF-STRING  (SEVR:INVALID STAT:UDF)
CTT:MOTOR 0.000 deg. 
CTT:MOTOR.VELO 0.000 deg./se 
CTT:STATIC-LONGIN 20 test eg (SEVR:INVALID STAT:UDF)
CTT:STATIC-AO 15 test eg 
CTT:STATIC-AO.OUT "CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest NPP NMS" 
CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest.NAME "CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest" (SEVR:INVALID STAT:UDF)

caGets
CTT:TEST_1-AI.SCAN ".1 second" 
CTT:TEST_1-AI.INP "CTT:TEST_1 NPP NMS" 
CTT:TEST_1-AI.EGU "ds" 
CTT:CALC_BI.INAV "Local PV" 
CTT:TEST_MBBO "50.00 Hz" (SEVR:MAJOR STAT:STATE)
CTT:TEST_MBBO.ZRVL 1 (SEVR:MAJOR STAT:STATE)
CTT:TEST_MBBO.DESC "Select RF 1 event freq" (SEVR:MAJOR STAT:STATE)
CTT:TEST_STRING "Hello world" 
CTT:STRING "Value field can have only fourty chars." (SEVR:INVALID STAT:LINK)
CTT:STRING.OUT "ten1234567twenty1234thirty1234forty12345fifty12345sixty12345 NPP NMS" (SEVR:INVALID STAT:LINK)
CTT:STRING.DESC "Description field can have only fourty ." (SEVR:INVALID STAT:LINK)
CTT:TEST_WVF-CHAR  
CTT:TEST_WVF-DOUBLE  
CTT:TEST_WVF-FLOAT  
CTT:TEST_WVF-FLOAT.NELM 10 
CTT:TEST_WVF-LONG  
CTT:TEST_WVF-SHORT  
CTT:TEST_WVF-STRING  
CTT:MOTOR 0.000 deg. 
CTT:MOTOR.VELO 0.000 deg./se 
CTT:STATIC-LONGIN 20 test eg 
CTT:STATIC-AO 15 test eg 
CTT:STATIC-AO.OUT "CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest NPP NMS" 
CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest.NAME "CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest" 

caDo

caPut
CTT:TEST_1-AI.SCAN ".5 second" 
CTT:TEST_1-AI.EGU "tooLongEguToWri" 
CTT:TEST_MBBO "10.00 Hz" 
CTT:TEST_MBBO.ZRVL 0 
CTT:TEST_MBBO.DESC "CTT description" 
CTT:STRING "Value field can have only fourty chars_" (SEVR:INVALID STAT:LINK)
Warning: Input ten1234567twenty1234thirty1234forty12345fifty12345sixty1234_ is longer than the allowed size 39.
Warning: Truncating to: ten1234567twenty1234thirty1234forty1234
CTT:STRING.OUT "ten1234567twenty1234thirty1234forty1234" (SEVR:INVALID STAT:LINK)
CTT:TEST_WVF-CHAR "a 1 b 2 c 3" 
CTT:TEST_WVF-DOUBLE 1 2 100 10 
CTT:TEST_WVF-DOUBLE.PREC 5 
CTT:TEST_WVF-FLOAT 1 2 100 10 
CTT:TEST_WVF-LONG 1 2 100 10 6 7 8 9 10 11 
CTT:TEST_WVF-SHORT 1 100 1000 10000 -31072 16960 
CTT:TEST_WVF-STRING "1st" "string" "bring" "dring" "ping" 
Error: CTT:MOTOR write response timed-out
CTT:MOTOR 5.000 deg. 
CTT:MOTOR.VELO 8.000 deg./se 
CTT:STATIC-LONGIN 0 test eg 
CTT:STATIC-AO 0 test eg 

caPutq
Warning: Input ten1234567twenty1234thirty1234forty12345fifty12345sixty1234! is longer than the allowed size 39.
Warning: Truncating to: ten1234567twenty1234thirty1234forty1234

caGet
CTT:TEST_1-AI.SCAN ".2 second" 
CTT:TEST_1-AI.INP "CTT:TEST_1 NPP NMS" 
CTT:TEST_1-AI.EGU "ohTooLongEguToW" 
CTT:CALC_BI.INAV "Local PV" 
CTT:TEST_MBBO "16.66 Hz" 
CTT:TEST_MBBO.ZRVL 1 
CTT:TEST_MBBO.DESC "CTTest description" 
CTT:TEST_STRING "Hello world" 
CTT:STRING "Value field can have only fourty chars!" (SEVR:INVALID STAT:LINK)
CTT:STRING.OUT "ten1234567twenty1234thirty1234forty1234 NPP NMS" (SEVR:INVALID STAT:LINK)
CTT:STRING.DESC "Description field can have only fourty ." (SEVR:INVALID STAT:LINK)
CTT:TEST_WVF-CHAR "1 a 2 b 3 c" 
CTT:TEST_WVF-DOUBLE 2.00000 2.30000 100.00000 10.00000 
CTT:TEST_WVF-FLOAT 2 2 100 10 
CTT:TEST_WVF-FLOAT.NELM 10 
CTT:TEST_WVF-LONG 2 2 100 10 6 7 8 9 10 11 
CTT:TEST_WVF-SHORT 2 100 1000 10000 -31072 16960 
CTT:TEST_WVF-STRING "2nd" "string" "bring" "dring" "ping" 
CTT:MOTOR 4.000 deg. 
CTT:MOTOR.VELO 5.000 deg./se 
CTT:STATIC-LONGIN 1 test eg 
CTT:STATIC-AO 1 test eg 
CTT:STATIC-AO.OUT "CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest NPP NMS" 
CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest.NAME "CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest" 

caGets
CTT:TEST_1-AI.SCAN ".2 second" 
CTT:TEST_1-AI.INP "CTT:TEST_1 NPP NMS" 
CTT:TEST_1-AI.EGU "ohTooLongEguToW" 
CTT:CALC_BI.INAV "Local PV" 
CTT:TEST_MBBO "16.66 Hz" 
CTT:TEST_MBBO.ZRVL 1 
CTT:TEST_MBBO.DESC "CTTest description" 
CTT:TEST_STRING "Hello world" 
CTT:STRING "Value field can have only fourty chars!" (SEVR:INVALID STAT:LINK)
CTT:STRING.OUT "ten1234567twenty1234thirty1234forty1234 NPP NMS" (SEVR:INVALID STAT:LINK)
CTT:STRING.DESC "Description field can have only fourty ." (SEVR:INVALID STAT:LINK)
CTT:TEST_WVF-CHAR "1 a 2 b 3 c" 
CTT:TEST_WVF-DOUBLE 2.00000 2.30000 100.00000 10.00000 
CTT:TEST_WVF-FLOAT 2 2 100 10 
CTT:TEST_WVF-FLOAT.NELM 10 
CTT:TEST_WVF-LONG 2 2 100 10 6 7 8 9 10 11 
CTT:TEST_WVF-SHORT 2 100 1000 10000 -31072 16960 
CTT:TEST_WVF-STRING "2nd" "string" "bring" "dring" "ping" 
Error: CTT:MOTOR.PROC write response timed-out
CTT:MOTOR 4.000 deg. 
Error: CTT:MOTOR.PROC write response timed-out
CTT:MOTOR.VELO 5.000 deg./se 
CTT:STATIC-LONGIN 1 test eg 
CTT:STATIC-AO 1 test eg 
CTT:STATIC-AO.OUT "CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest NPP NMS" 
CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest.NAME "CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest" 

caDo

-----------------------------------------
Various output formatting
-----------------------------------------
Using: -plain
".2 second" 
"CTT:TEST_1 NPP NMS" 
"ohTooLongEguToW" 
"Local PV" 
"16.66 Hz" 
1.000000 
"CTTest description" 
"Hello world" 
"Value field can have only fourty chars!" 
"ten1234567twenty1234thirty1234forty1234 NPP NMS" 
"Description field can have only fourty ." 
"1 a 2 b 3 c" 
2.000000 2.300000 100.000000 10.000000 
2.000000 2.300000 100.000000 10.000000 
10.000000 
2 2 100 10 6 7 8 9 10 11 
2 100 1000 10000 -31072 16960 
"2nd" "string" "bring" "dring" "ping" 
4.000000 
5.000000 
1 
1.000000 
"CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest NPP NMS" 
"CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest" 

Using: -noname -nostat -nounit
".2 second" 
"CTT:TEST_1 NPP NMS" 
"ohTooLongEguToW" 
"Local PV" 
"16.66 Hz" 
1 
"CTTest description" 
"Hello world" 
"Value field can have only fourty chars!" 
"ten1234567twenty1234thirty1234forty1234 NPP NMS" 
"Description field can have only fourty ." 
"1 a 2 b 3 c" 
2.00000 2.30000 100.00000 10.00000 
2 2 100 10 
10 
2 2 100 10 6 7 8 9 10 11 
2 100 1000 10000 -31072 16960 
"2nd" "string" "bring" "dring" "ping" 
4.000 
5.000 
1 
1 
"CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest NPP NMS" 
"CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest" 

Using: -stat -plain
Warning: -plain option overrides all output formatting switches (except date/time related).
".2 second" 
Warning: -plain option overrides all output formatting switches (except date/time related).
"CTT:TEST_1 NPP NMS" 
Warning: -plain option overrides all output formatting switches (except date/time related).
"ohTooLongEguToW" 
Warning: -plain option overrides all output formatting switches (except date/time related).
"Local PV" 
Warning: -plain option overrides all output formatting switches (except date/time related).
"16.66 Hz" 
Warning: -plain option overrides all output formatting switches (except date/time related).
1.000000 
Warning: -plain option overrides all output formatting switches (except date/time related).
"CTTest description" 
Warning: -plain option overrides all output formatting switches (except date/time related).
"Hello world" 
Warning: -plain option overrides all output formatting switches (except date/time related).
"Value field can have only fourty chars!" 
Warning: -plain option overrides all output formatting switches (except date/time related).
"ten1234567twenty1234thirty1234forty1234 NPP NMS" 
Warning: -plain option overrides all output formatting switches (except date/time related).
"Description field can have only fourty ." 
Warning: -plain option overrides all output formatting switches (except date/time related).
"1 a 2 b 3 c" 
Warning: -plain option overrides all output formatting switches (except date/time related).
2.000000 2.300000 100.000000 10.000000 
Warning: -plain option overrides all output formatting switches (except date/time related).
2.000000 2.300000 100.000000 10.000000 
Warning: -plain option overrides all output formatting switches (except date/time related).
10.000000 
Warning: -plain option overrides all output formatting switches (except date/time related).
2 2 100 10 6 7 8 9 10 11 
Warning: -plain option overrides all output formatting switches (except date/time related).
2 100 1000 10000 -31072 16960 
Warning: -plain option overrides all output formatting switches (except date/time related).
"2nd" "string" "bring" "dring" "ping" 
Warning: -plain option overrides all output formatting switches (except date/time related).
4.000000 
Warning: -plain option overrides all output formatting switches (except date/time related).
5.000000 
Warning: -plain option overrides all output formatting switches (except date/time related).
1 
Warning: -plain option overrides all output formatting switches (except date/time related).
1.000000 
Warning: -plain option overrides all output formatting switches (except date/time related).
"CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest NPP NMS" 
Warning: -plain option overrides all output formatting switches (except date/time related).
"CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest" 

Using: -stat -nostat
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:TEST_1-AI.SCAN ".2 second" (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:TEST_1-AI.INP "CTT:TEST_1 NPP NMS" (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:TEST_1-AI.EGU "ohTooLongEguToW" (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:CALC_BI.INAV "Local PV" (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:TEST_MBBO "16.66 Hz" (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:TEST_MBBO.ZRVL 1 (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:TEST_MBBO.DESC "CTTest description" (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:TEST_STRING "Hello world" (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:STRING "Value field can have only fourty chars!" (SEVR:INVALID STAT:LINK)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:STRING.OUT "ten1234567twenty1234thirty1234forty1234 NPP NMS" (SEVR:INVALID STAT:LINK)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:STRING.DESC "Description field can have only fourty ." (SEVR:INVALID STAT:LINK)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:TEST_WVF-CHAR "1 a 2 b 3 c" (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:TEST_WVF-DOUBLE 2.00000 2.30000 100.00000 10.00000 (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:TEST_WVF-FLOAT 2 2 100 10 (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:TEST_WVF-FLOAT.NELM 10 (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:TEST_WVF-LONG 2 2 100 10 6 7 8 9 10 11 (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:TEST_WVF-SHORT 2 100 1000 10000 -31072 16960 (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:TEST_WVF-STRING "2nd" "string" "bring" "dring" "ping" (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:MOTOR 4.000 deg. (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:MOTOR.VELO 5.000 deg./se (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:STATIC-LONGIN 1 test eg (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:STATIC-AO 1 test eg (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:STATIC-AO.OUT "CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest NPP NMS" (SEVR:NO_ALARM STAT:NO_ALARM)
Warning: Options -stat and -nostat are mutually exclusive. Using -stat.
CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest.NAME "CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest" (SEVR:NO_ALARM STAT:NO_ALARM)

Using: -d -t -localdate -localtime
server time: 2016-09-20 18:38:06.393141 local time: 2016-09-20 18:38:09.405951 CTT:TEST_1-AI.SCAN ".2 second" 
server time: 2016-09-20 18:38:06.393141 local time: 2016-09-20 18:38:09.451189 CTT:TEST_1-AI.INP "CTT:TEST_1 NPP NMS" 
server time: 2016-09-20 18:38:06.393141 local time: 2016-09-20 18:38:09.495980 CTT:TEST_1-AI.EGU "ohTooLongEguToW" 
server time: 2016-09-20 18:38:06.393107 local time: 2016-09-20 18:38:09.545895 CTT:CALC_BI.INAV "Local PV" 
server time: 2016-09-20 18:38:00.719613 local time: 2016-09-20 18:38:09.595353 CTT:TEST_MBBO "16.66 Hz" 
server time: 2016-09-20 18:38:00.719613 local time: 2016-09-20 18:38:09.638173 CTT:TEST_MBBO.ZRVL 1 
server time: 2016-09-20 18:38:00.719613 local time: 2016-09-20 18:38:09.682151 CTT:TEST_MBBO.DESC "CTTest description" 
server time: 2016-09-20 18:38:00.783579 local time: 2016-09-20 18:38:09.726355 CTT:TEST_STRING "Hello world" 
server time: 2016-09-20 18:37:57.082552 local time: 2016-09-20 18:38:09.770760 CTT:STRING "Value field can have only fourty chars!" (SEVR:INVALID STAT:LINK)
server time: 2016-09-20 18:37:57.082552 local time: 2016-09-20 18:38:09.846240 CTT:STRING.OUT "ten1234567twenty1234thirty1234forty1234 NPP NMS" (SEVR:INVALID STAT:LINK)
server time: 2016-09-20 18:37:57.082552 local time: 2016-09-20 18:38:09.921084 CTT:STRING.DESC "Description field can have only fourty ." (SEVR:INVALID STAT:LINK)
server time: 2016-09-20 18:38:00.847416 local time: 2016-09-20 18:38:09.968692 CTT:TEST_WVF-CHAR "1 a 2 b 3 c" 
server time: 2016-09-20 18:38:00.911094 local time: 2016-09-20 18:38:10.014227 CTT:TEST_WVF-DOUBLE 2.00000 2.30000 100.00000 10.00000 
server time: 2016-09-20 18:38:01.038714 local time: 2016-09-20 18:38:10.060169 CTT:TEST_WVF-FLOAT 2 2 100 10 
server time: 2016-09-20 18:38:01.038714 local time: 2016-09-20 18:38:10.105908 CTT:TEST_WVF-FLOAT.NELM 10 
server time: 2016-09-20 18:38:01.103176 local time: 2016-09-20 18:38:10.153795 CTT:TEST_WVF-LONG 2 2 100 10 6 7 8 9 10 11 
server time: 2016-09-20 18:38:01.166884 local time: 2016-09-20 18:38:10.202434 CTT:TEST_WVF-SHORT 2 100 1000 10000 -31072 16960 
server time: 2016-09-20 18:38:01.230647 local time: 2016-09-20 18:38:10.246204 CTT:TEST_WVF-STRING "2nd" "string" "bring" "dring" "ping" 
server time: 2016-09-20 18:38:01.358202 local time: 2016-09-20 18:38:10.290052 CTT:MOTOR 4.000 deg. 
server time: 2016-09-20 18:38:01.358202 local time: 2016-09-20 18:38:10.334407 CTT:MOTOR.VELO 5.000 deg./se 
server time: 2016-09-20 18:38:01.422086 local time: 2016-09-20 18:38:10.379527 CTT:STATIC-LONGIN 1 test eg 
server time: 2016-09-20 18:38:01.551863 local time: 2016-09-20 18:38:10.424757 CTT:STATIC-AO 1 test eg 
server time: 2016-09-20 18:38:01.551863 local time: 2016-09-20 18:38:10.501171 CTT:STATIC-AO.OUT "CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest NPP NMS" 
server time: 2016-09-20 18:38:00.230483 local time: 2016-09-20 18:38:10.576299 CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest.NAME "CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest" 

Using: -mukmuk
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'
../caget: unrecognized option '-mukmuk'

-----------------------------------------
caPut on waveforms
-----------------------------------------
with -bin
CTT:TEST_WVF-CHAR 0b00000001 0b00000010 0b00000101 
CTT:TEST_WVF-DOUBLE 0b00000000000000000000000000000001 0b00000000000000000000000000001010 0b00000000000000000000000001100101 
CTT:TEST_WVF-FLOAT 0b00000000000000000000000000000001 0b00000000000000000000000000001010 0b00000000000000000000000001100101 
CTT:TEST_WVF-LONG 0b00000000000000000000000000000001 0b00000000000000000000000000000010 0b00000000000000000000000000000101 
CTT:TEST_WVF-SHORT 0b0000000000000001 0b0000000000000010 0b0000000000000101 
CTT:TEST_WVF-STRING "123" "string" "456" "dring" "ping" 

with -oct
CTT:TEST_WVF-CHAR 0o1 0o10 0o101 
CTT:TEST_WVF-DOUBLE 0o1 0o12 0o145 
CTT:TEST_WVF-FLOAT 0o1 0o12 0o145 
CTT:TEST_WVF-LONG 0o1 0o10 0o101 
CTT:TEST_WVF-SHORT 0o1 0o10 0o101 
CTT:TEST_WVF-STRING "123" "string" "456" "dring" "ping" 

with -hex
Error: 101 can not be parsed as an 8 bit integer using numeric base 16
Error: Problem parsing input for PV CTT:TEST_WVF-CHAR: "1 10 101".
CTT:TEST_WVF-DOUBLE 0x1 0xa 0x65 
CTT:TEST_WVF-FLOAT 0x1 0xa 0x65 
CTT:TEST_WVF-LONG 0x1 0x10 0x101 
CTT:TEST_WVF-SHORT 0x1 0x10 0x101 
CTT:TEST_WVF-STRING "123" "string" "456" "dring" "ping" 

-----------------------------------------
Test setting and reading enum values with number or string
-----------------------------------------
CTT:TEST_MBBO "50.00 Hz" (SEVR:MAJOR STAT:STATE)
CTT:TEST_MBBO "25.00 Hz" 
CTT:TEST_MBBO 3 
CTT:TEST_MBBO "25.00 Hz" 
CTT:BO "1" 
CTT:BO "0" 
CTT:BO "1" 
CTT:BO "0" 
CTT:BO "1" 

-----------------------------------------
Test setting and reading chars as numbers or strings
-----------------------------------------
CTT:TEST_WVF-CHAR "1 2" 
CTT:TEST_WVF-CHAR 49 32 50 0 
CTT:TEST_WVF-CHAR "1 2" 
CTT:TEST_WVF-CHAR 1 2 
CTT:TEST_WVF-CHAR 1 2 
CTT:TEST_WVF-CHAR 1 2 
CTT:TEST_WVF-CHAR 50 32 32 32 49 
CTT:TEST_WVF-CHAR 50 32 32 32 49 
CTT:TEST_WVF-CHAR "2   1" 

-----------------------------------------
Double conversion
-----------------------------------------
CTT:CALC 10 egu (SEVR:INVALID STAT:UDF)
CTT:CALC.PREC 4 egu (SEVR:INVALID STAT:UDF)
CTT:CALC 0xa egu (SEVR:INVALID STAT:UDF)
CTT:CALC 0b00000000000000000000000000001010 egu (SEVR:INVALID STAT:UDF)
CTT:CALC 0o12 egu (SEVR:INVALID STAT:UDF)
CTT:CALC 10.2300 egu (SEVR:INVALID STAT:UDF)
CTT:CALC 10.2300 egu (SEVR:INVALID STAT:UDF)

-----------------------------------------
Number of elements
-----------------------------------------
CTT:TEST_WVF-CHAR 5 "2   1" 
CTT:TEST_WVF-DOUBLE 3 1.00000 10.00000 101.00000 
CTT:TEST_WVF-FLOAT 3 1 10 101 
CTT:TEST_WVF-LONG 3 1 16 257 
CTT:TEST_WVF-SHORT 3 1 16 257 
CTT:TEST_WVF-STRING 5 "123" "string" "456" "dring" "ping" 
CTT:PROC-CALC 1 2 
CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest.NAME 1 "CTT:RecordNameThatIsLongerThanFortyCharsForLongStringTest" 
